girder-dandi-archive/girder_dandi_archive/schema_validation.py

```diff
@@ -0,0 +1,21 @@
+from girder.exceptions import ValidationException
+
+from .constants import SCHEMA_MINIMUM_VERSION
+
+
+def validate_schema(event):
+    meta = event.info["meta"]
+    if "dandiset" not in meta:
+        # Do not validate if there is no dandiset metadata property
+        # Only top level dandiset folders should be validated
+        return
+    dandiset = meta["dandiset"]
+    if "schemaVersion" not in dandiset:
+        raise ValidationException("No schema version specified on folder")
+    schema_version = dandiset["schemaVersion"]
+    # TODO this will break if MINIMUM_VERSION ever has 2-digit numbers
+    if schema_version < SCHEMA_MINIMUM_VERSION:
```

Comment:
I believe plain string comparison wouldn't work here since  
```
$> python -c 'print("1.0.0-rc1" < "1.0.0")'
False
```
I use `semantic-version` package in dandi-cli:
```
$> python3 -c 'from semantic_version import Version; print(Version("1.0.0-rc1") < Version("1.0.0"))'
True
```
but please add additional check/dedicated raise to first check if provided version is "legit semver":
```
$> python3 -c 'from semantic_version import Version; print(Version("1.0.0kukaracha") < Version("1.0.0"))'   
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/semantic_version/base.py", line 105, in __init__
    major, minor, patch, prerelease, build = self.parse(version_string, partial)
  File "/usr/lib/python3/dist-packages/semantic_version/base.py", line 303, in parse
    raise ValueError('Invalid version string: %r' % version_string)
ValueError: Invalid version string: '1.0.0kukaracha'

$> python3 -c 'from semantic_version import validate, Version; print(validate("1.0.0kukaracha"))'                     
False
```